compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -522,9 +522,18 @@ pub fn check_crate(krate: &ast::Crate, sess: &Session, features: &Features) {
         "consider removing `for<...>`"
     );
     gate_all!(more_qualified_paths, "usage of qualified paths in this context is experimental");
-    for &span in spans.get(&sym::yield_expr).iter().copied().flatten() {
-        if !span.at_least_rust_2024() {
-            gate!(&visitor, coroutines, span, "yield syntax is experimental");
+    // yield can be enabled either by `coroutines` or `gen_blocks`
+    if let Some(spans) = spans.get(&sym::yield_expr) {
+        for span in spans {
+            if (!visitor.features.coroutines() && !span.allows_unstable(sym::coroutines))
+                && (!visitor.features.gen_blocks() && !span.allows_unstable(sym::gen_blocks))
+            {
+                #[allow(rustc::untranslatable_diagnostic)] // FIXME: make this translatable
```

Comment:
```suggestion
```
You don't need to do this, see #132181

Reply:
Hm, I get an error if it is removed:

```
error: diagnostics should be created using translatable messages
   --> compiler/rustc_ast_passes/src/feature_gate.rs:534:68
    |
534 |                 feature_err(&visitor.sess, sym::coroutines, *span, "yield syntax is experimental")
    |                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[deny(rustc::untranslatable_diagnostic)]` on by default
```

so I'm a little confused.

Reply:
Oh right, it must be in a crate that has a `deny` even though these are now allow-by-default - feel free to keep the `allow` but remove the `FIXME`.

Reply:
Sure, done!